samples/client/petstore/rust/reqwest/fileResponseTest/src/apis/default_api.rs

```diff
@@ -47,7 +47,16 @@ impl DefaultApi for DefaultApiClient {
         }
 
         let req = req_builder.build()?;
-        Ok(client.execute(req)?.error_for_status()?.json()?)
+        let mut resp = client.execute(req)?;
+        if resp.status().is_success() {
+            Ok(resp.json()?)
+        } else {
+            let status = resp.status();
+            let content = resp.text()?;
+            let entity: Option<serde_json::Value> = serde_json::from_str(&content).ok();
+            let error = crate::apis::ResponseErrorContent { status, content, entity };
+            Err(Error::ResponseError(error))
+        }
```

Comment:
hey :raising_hand_man:  great attempt on this issue, but this isn't complete yet.

The openapi spec allows for defining a schema for every possible response status: eg

```json
"responses": {
    "200": {},
    "202": {},
    "400": {},
    "default": {}
}
```

so instead of just checking for a success status, you'll need to match over the response status.

and deserialize based on the defined schema for that status as well as the fallback(default) schema.

I imagine this means that the return type of methods would become an enum, 

```rust
pub enum OperationIdResponse {
    Status200(Status200Schema),
    Status202(Status202Schema),
    Status400(Status400Schema),
    Unspecified(DefaultSchema)
}
```

Reply:
Hi @seunlanlege,

Thanks for your comment. I know this is just a first attempt which is incomplete.

I had the same idea of an enum as response, but I don't like it. I think we should not return `Ok` when the return status is not ok…

Reply:
I am working on something like:

```rust
#[derive(Debug, Clone)]
pub struct ResponseErrorContent<T> {
    pub status: reqwest::StatusCode,
    pub content: String,
    pub entity: Option<T>,
}

#[derive(Debug)]
pub enum Error<T> {
    Reqwest(reqwest::Error),
    Serde(serde_json::Error),
    Io(std::io::Error),
    ResponseError(ResponseErrorContent<T>),
}
```

where T could be an enum of error types corresponding to error codes.

Reply:
also bear in mind that there could be more than one success response in the `2xx` range, So success types would potentially be an enum

Reply:
With an enum, multiple success codes can be handled, but this case is not frequent. So I don't know if it is worth complicating the generated code for this.

Reply:
> multiple success codes can be handled, but this case is not frequent. 
unfortunately it's actually very common :sweat_smile: 

jira/sendgrid's openapi specifications have multiple 2xx success schemas